api/v1alpha4/clusterclass_webhook.go

```diff
@@ -261,7 +260,60 @@ func (r LocalObjectTemplate) validate(namespace string, pathPrefix *field.Path)
 			field.Invalid(
 				pathPrefix.Child("ref", "apiVersion"),
 				r.Ref.APIVersion,
-				"cannot be empty",
+				"value cannot be empty",
+			),
+		)
+	}
+
+	return allErrs
+}
+
+func (r LocalObjectTemplate) isCompatibleWith(old LocalObjectTemplate, pathPrefix *field.Path) field.ErrorList {
```

Comment:
Just wanted to note that r.ref can be nil. We already check this in `isValid` but as we still call `isCompatibleWith` independent of the results of `isValid`,  `isCompatibleWith` would panic in this case.

Reply:
This was documented in the comment inside the func, however I implement checks to avoid panic

Reply:
Absolutely, my comment was only about the panic :)

Reply:
I think we can stay in the current variant. we can eventually iterate on deducing error messages at later stage.